src/craftdef.h

```diff
@@ -44,6 +44,19 @@ enum CraftMethod
 };
 
 /*
+	The type a hash can be. The earlier a type is mentioned in this enum,
+	the earlier it is tried, and the less likely is a collision
+ */
+enum HashType
+{
+	HASH_TYPE_ITEM_NAMES = 0,
+	HASH_TYPE_COUNT,
+	HASH_TYPE_UNHASHED
+};
+const int hash_type_max = (int) HASH_TYPE_UNHASHED;
```

Comment:
I understand why he has done it... perhaps the enum should have HASH_TYPE_MAX (or _LAST) as the last item

Technically, if going the const route hash_type_max should be HASH_TYPE_COUNT, or HASH_TYPE_UNHASHED-1 since counting starts at 0